datafusion/core/tests/sql/timestamp.rs

```diff
@@ -1035,13 +1035,13 @@ async fn timestamp_sub_with_tz() -> Result<()> {
     let sql = "SELECT val, ts1 - ts2 AS ts_diff FROM table_a ORDER BY ts2 - ts1";
     let actual = execute_to_batches(&ctx, sql).await;
     let expected = vec![
-        "+-----+---------------------------------------------------+",
-        "| val | ts_diff                                           |",
-        "+-----+---------------------------------------------------+",
-        "| 3   | 0 years 0 mons 0 days 10 hours 0 mins 30.000 secs |",
-        "| 1   | 0 years 0 mons 0 days 10 hours 0 mins 20.000 secs |",
-        "| 2   | 0 years 0 mons 0 days 10 hours 0 mins 10.000 secs |",
-        "+-----+---------------------------------------------------+",
+        "+-----+---------+",
+        "| val | ts_diff |",
+        "+-----+---------+",
+        "| 3   | PT30S   |",
```

Comment:
Timestamp arithmetic now returns the appropriate duration type

Reply:
I would be happy to change the formatting upstream if there is consensus on what it should be. We are currently just doing what chrono does, which appears to be [ISO 8601](https://en.wikipedia.org/wiki/ISO_8601#Time_intervals)

Reply:
i think the coercion rules might not work well. 
```bash
❯ select ('2023-01-02'::timestamp - '2023-01-01'::timestamp)::interval;
+--------------------------------------------------------+
| Utf8("2023-01-02") - Utf8("2023-01-01")                |
+--------------------------------------------------------+
| 0 years 0 mons 0 days 24 hours 0 mins 0.000000000 secs |
+--------------------------------------------------------+
1 row in set. Query took 0.002 seconds.
```
it's converted to (0 month, 0 day, 86400000000000 nanos) instead of (0 month, 1 day, 0 nanos)

which isn't consist with 
```bash
❯ select ('2023-01-02'::timestamp - '2023-01-01'::timestamp);
+-----------------------------------------+
| Utf8("2023-01-02") - Utf8("2023-01-01") |
+-----------------------------------------+
| P1D                                     |
+-----------------------------------------+
1 row in set. Query took 0.002 seconds.
``` 

both formats are fine so we. in case we decide to align with interval format, we should discuss whether it outputs 
`0 years 0 mons 1 days 0 hours 0 mins 0 secs` or 
`0 years 0 mons 0 days 24 hours 0 mins 0 secs` in this case

Reply:
I think we should probably change duration output to be a number of the duration unit, i.e. seconds, milliseconds, etc... to avoid potential ambiguity over what the duration represents. It might be confusing if both intervals and durations report some number of days, but arithmetic involving them has different semantics across DST boundaries

Reply:
I have filed two tickets of interest: 
* https://github.com/apache/arrow-datafusion/issues/7068 which tries to describes the change in semantics this PR does
* https://github.com/apache/arrow-datafusion/issues/7070 that proposes special casing the display of Duration, based on @waitingkuo 's feedback above